hit.ml

```diff
@@ -122,7 +123,39 @@ let doc_ ?(id=true) ?found ?highlight ?fields source =
 *)
 let doc_no_source ?fields () = doc_ ~found:(Simple Bool) ?fields None
 let doc source = doc_ ~found:(Simple Bool) (Some source)
-let hit ?highlight ?id ?fields source = doc_ ?highlight ?id ?fields (Some source)
+let hit ?highlight ?id ?fields ?inner_hits source = doc_ ?highlight ?id ?fields ?inner_hits (Some source)
+
+let nested_meta () =
+  Dict [
+    "field", Simple String;
+    "offset", Simple Int;
+  ]
+
+let inner_hit ~nested ~highlight ?fields source =
+  let source_type = get_nested nested source in
+  let a = [
+    "_id", Simple String;
+    "_nested", nested_meta ();
+    "_source", source_type;
+  ] @
+  (match highlight with None -> [] | Some h -> ["highlight", h]) @
+  (match fields with None -> [] | Some f -> ["fields", f])
+  in
```

Comment:
```suggestion
  let a = 
    (match highlight with None -> [] | Some h -> ["highlight", h]) @
    (match fields with None -> [] | Some f -> ["fields", f])
    @ [
    "_id", Simple String;
    "_nested", nested_meta ();
    "_source", source_type;
  ]
  in
```

splitting hair a bit here, but for perf it's better to have the small list on the left side of the concat